test/put.sh

```diff
@@ -293,6 +293,33 @@ it_can_put_and_set_git_config() {
   mv ~/.gitconfig.orig ~/.gitconfig
 }
 
+it_can_put_and_force_the_push() {
+  local repo1=$(init_repo)
+
+  local src=$(mktemp -d $TMPDIR/put-src.XXXXXX)
+  local repo2=$src/repo
+  git clone $repo1 $repo2
+
+  local ref=$(make_commit $repo2)
+  make_commit $repo1
+  # cannot push to repo while it's checked out to a branch
+  git -C $repo1 checkout refs/heads/master
+
+  cp ~/.gitconfig ~/.gitconfig.orig
+
+  put_uri_with_force $repo1 $src repo | jq -e "
+    .version == {ref: $(echo $ref | jq -R .)}
+  "
+
+  # switch back to master
+  git -C $repo1 checkout master
+
+  test "$(git config --global core.pager)" == 'true'
```

Comment:
don't need these here - some of the rest might also be copypasta. maybe just base it on the simplest push case?